### PR TITLE
[DOC] Add docstrings for some bigger private functions in `load_confounds.`

### LINKED ISSUE #3826: [DOC] `load_confounds` private function docstrings
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe your proposed suggestion in detail.

I feel the need to complete the docstrings of some private functions. 
Some private functions are complex enough that need docstrings for contributors to understand the logics.
And perhaps some typos.

### List any pages that would be impacted.

None. Just private functions.

### PR DESCRIPTION
**Pull request by htwangtw**
Closes #3826.

- Focus on some wrapper functions that I have seen used incorrectly from user submitted issues.
- With the help of copilot.

### CODE REVIEW

File: nilearn/interfaces/fmriprep/load_confounds.py [hunk 1]
```diff
@@ -315,7 +327,31 @@ def load_confounds(
...
+    demean : boolean, default True
+        See :func:`nilearn.interfaces.fmriprep.load_confounds` for details.
+
+    kwargs : dict
+        Extra relevant parameters for the given `strategy`.
+        See :func:`nilearn.interfaces.fmriprep.load_confounds` for details.
+
+    Returns
+    -------
+    sample_mask : None, numpy.ndarray, or list of
```
Comment (Remi-Gau) [inline nilearn/interfaces/fmriprep/load_confounds.py:349]: just to be sure this means :
- None, numpy.ndarray, or list of numpy.ndarray

Comment (bthirion) [reply nilearn/interfaces/fmriprep/load_confounds.py:349]:
```suggestion
    sample_mask : None, :class:`numpy.ndarray`, or list of
```

File: nilearn/interfaces/fmriprep/load_confounds.py [hunk 2]
```diff
@@ -315,7 +327,31 @@ def load_confounds(
...
+
+    Parameters
+    ----------
+    image_file : str
+        Path to processed image file.
+
+    strategy : tuple or list of strings.
+        See :func:`nilearn.interfaces.fmriprep.load_confounds` for details.
+
+    demean : boolean, default True
```
Comment (bthirion) [inline nilearn/interfaces/fmriprep/load_confounds.py:340]:
```suggestion
    demean : :obj:`bool`,, default True
```

File: nilearn/interfaces/fmriprep/load_confounds.py [hunk 3]
```diff
@@ -338,7 +374,37 @@ def _load_confounds_for_single_image_file(
...
+
+    Parameters
+    ----------
+    confounds_file : str
+        Path to confounds file.
+
+    strategy : tuple or list of strings.
+        See :func:`nilearn.interfaces.fmriprep.load_confounds` for details.
+
+    demean : boolean, default True
```
Comment (bthirion) [inline nilearn/interfaces/fmriprep/load_confounds.py:387]:
```suggestion
    demean : :obj:`bool`,, default True
```

**Review by bthirion — COMMENTED** [review]
Thx !
Maybe we could take this opportunity to improve docstrings quality...

File: nilearn/interfaces/fmriprep/load_confounds.py (hunk 1 above)
Comment (htwangtw) [reply nilearn/interfaces/fmriprep/load_confounds.py:349]: > just to be sure. 
> This means : "None, numpy.ndarray, or list of numpy.ndarray" ?

None, numpy.ndarray, or list of all of the above.
When the input is a list of images, the output may or may not have a sample mask depending on whether they need scrubbing or not etc

What would be a better way to explain this?

Comment (Remi-Gau) [reply nilearn/interfaces/fmriprep/load_confounds.py:349]:
```
 None, :class:`numpy.ndarray` or, :obj:`list` of :class:`numpy.ndarray` or None
```

Would this work?

File: nilearn/interfaces/fmriprep/load_confounds_compcor.py [hunk 4]
```diff
@@ -38,7 +55,22 @@ def _find_compcor(confounds_json, compcor, n_compcor):
...
 def _select_compcor(compcor_cols, n_compcor):
-    """Retain a specified number of compcor components."""
+    """Retain a specified number of compcor components.
+
+    Parameters
+    ----------
+    compcor_cols : list
+        List of compcor components column names filtered by user
+        selected CompCor strategy.
+    n_compcor : int or str
```
Comment (bthirion) [inline nilearn/interfaces/fmriprep/load_confounds_compcor.py:65]: Note: to make the APi simpler, one could encode 'all' with '-1'

File: nilearn/interfaces/fmriprep/load_confounds_compcor.py [hunk 5]
```diff
@@ -47,7 +79,21 @@ def _select_compcor(compcor_cols, n_compcor):
 
 
 def _check_compcor_method(compcor):
-    """Load compcor options and check if method is acceptable."""
+    """Load compcor options and check if method is acceptable.
+
+    Parameters
+    ----------
+    compcor : str
```
Comment (bthirion) [inline nilearn/interfaces/fmriprep/load_confounds_compcor.py:86]:
```suggestion
    compcor : :obj: `str`
```

File: nilearn/interfaces/fmriprep/load_confounds_compcor.py [hunk 6]
```diff
@@ -63,7 +109,26 @@ def _check_compcor_method(compcor):
 
 
 def _acompcor_mask(confounds_json, anat_mask, compcor_cols_filt, n_compcor):
-    """Filter according to acompcor mask(s) and select top components."""
+    """Filter according to acompcor mask(s) and select top components.
+
+    Parameters
+    ----------
+    confounds_json : dict
```
Comment (bthirion) [inline nilearn/interfaces/fmriprep/load_confounds_compcor.py:116]:
```suggestion
    confounds_json : :obj: `dict
```

File: nilearn/interfaces/fmriprep/load_confounds_compcor.py [hunk 7]
```diff
@@ -73,7 +138,24 @@ def _acompcor_mask(confounds_json, anat_mask, compcor_cols_filt, n_compcor):
...
 def _json_mask(compcor_cols_filt, confounds_json, mask):
-    """Extract anat compcor components from a given mask."""
+    """Extract anat compcor components with a given mask.
+
+    Parameters
+    ----------
+    compcor_cols_filt : list
+        List of compcor components column names filtered by user
+        selected CompCor strategy.
+    confounds_json : dict
```
Comment (bthirion) [inline nilearn/interfaces/fmriprep/load_confounds_compcor.py:148]: same here

File: nilearn/interfaces/fmriprep/load_confounds_compcor.py [hunk 8]
```diff
@@ -73,7 +138,24 @@ def _acompcor_mask(confounds_json, anat_mask, compcor_cols_filt, n_compcor):
...
+        Dictionary of confounds from the confounds.json file.
+    mask : str
```
Comment (bthirion) [inline nilearn/interfaces/fmriprep/load_confounds_compcor.py:150]: +here

**Review by bthirion — COMMENTED** [review]
Thx. A few changes have been missed. Best,

**Review by ymzayek — COMMENTED** [review]
@htwangtw thanks for adding the docstrings to these private functions! Most of my comments actually have to do with modified docstrings of public functions but I think it's fine to change them in this PR. If you think it's a bigger job (I didn't check parameters that aren't modified) and wish to leave it for another PR that's fine too

File: nilearn/interfaces/fmriprep/load_confounds_compcor.py (hunk 5 above)
Comment (htwangtw) [reply nilearn/interfaces/fmriprep/load_confounds_compcor.py:86]: These functions are not exposed in the documentation. Is it still worth doing it?

Comment (bthirion) [reply nilearn/interfaces/fmriprep/load_confounds_compcor.py]: No, indeed.

File: nilearn/interfaces/fmriprep/load_confounds_components.py [hunk 9]
```diff
@@ -38,7 +71,28 @@ def _load_high_pass(confounds_raw):
...
-    """Load the regressors derived from the white matter and CSF masks."""
+    """Load the regressors derived from the white matter and CSF masks.
+
+    Parameters
+    ----------
+    confounds_raw : pandas.DataFrame
+        DataFrame of confounds.
+
+    wm_csf : str
+        White matter and CSF strategy to use. Options are Options are "basic",
```
Comment (ymzayek) [inline nilearn/interfaces/fmriprep/load_confounds_components.py:82]:
```suggestion
        White matter and CSF strategy to use. Options are "basic",
```

File: nilearn/interfaces/fmriprep/load_confounds_components.py [hunk 10]
```diff
@@ -47,7 +101,28 @@ def _load_wm_csf(confounds_raw, wm_csf):
...
-    """Load the regressors derived from the global signal."""
+    """Load the regressors derived from the global signal.
+
+    Parameters
+    ----------
+    confounds_raw : pandas.DataFrame
+        DataFrame of confounds.
+
+    global_signal : str
+        Global signal strategy to use. Options are Options are "basic",
```
Comment (ymzayek) [inline nilearn/interfaces/fmriprep/load_confounds_components.py:112]:
```suggestion
        Global signal strategy to use. Options are "basic",
```

File: nilearn/interfaces/fmriprep/load_confounds_components.py [hunk 11]
```diff
@@ -13,7 +13,28 @@
...
-    """Load the motion regressors."""
+    """Load the motion regressors.
+
+    Parameters
+    ----------
+    confounds_raw : pandas.DataFrame
+        DataFrame of confounds.
+
+    motion : str
+        Motion strategy to use. Options are Options are "basic",
```
Comment (ymzayek) [inline nilearn/interfaces/fmriprep/load_confounds_components.py:24]:
```suggestion
        Motion strategy to use. Options are "basic",
```

File: nilearn/interfaces/fmriprep/load_confounds_utils.py [hunk 12]
```diff
@@ -35,7 +35,26 @@
...
+    params : list of str
+        List of parameters constructed based on users choices.
+
+    Returns
+    -------
+    bool
+        True if all parameters are found in the confounds.
+        False if none of the parameters are found in the confounds.
+        list of str
+            List of parameters that are not found in the confounds.
```
Comment (ymzayek) [inline nilearn/interfaces/fmriprep/load_confounds_utils.py:56]: Something went wrong here. Maybe something like:
```suggestion
    bool or list of str
        True if all parameters are found in the confounds.
        False if none of the parameters are found in the confounds.
        List of parameters that are not found in the confounds
        if only some parameters are found.
```

File: nilearn/interfaces/fmriprep/load_confounds_utils.py [hunk 13]
```diff
@@ -253,8 +419,29 @@ def _check_images(image_file, flag_full_aroma):
...
+    ----------
+    confounds : pandas.DataFrame
+        Confound regressors loaded based on user's choice.
+
+    demean : bool
+        True if the confounds should be demeaned, False otherwise.
+
+    Returns
+    -------
+    sample_mask : None numpy.ndarray
```
Comment (ymzayek) [inline nilearn/interfaces/fmriprep/load_confounds_utils.py:434]:
```suggestion
    sample_mask : None or numpy.ndarray
```

File: nilearn/interfaces/fmriprep/load_confounds_utils.py [hunk 14]
```diff
@@ -267,7 +454,27 @@ def _prepare_output(confounds, demean):
...
+    """Demean the confounds.
+
+    The mean is calculated on non-outlier values.
+
+    Parameters
+    ----------
+    confounds : pandas.DataFrame
+        Confound regressors loaded based on user's choice.
+
+    sample_mask : None numpy.ndarray
```
Comment (ymzayek) [inline nilearn/interfaces/fmriprep/load_confounds_utils.py:466]:
```suggestion
    sample_mask : None or numpy.ndarray
```

**Review by ymzayek — COMMENTED** [review]
Thx @htwangtw , just a few minor things

**Review by ymzayek — APPROVED** [review]
Thx!

**Review by Remi-Gau — APPROVED** [review]
Good with me

thanks @htwangtw

**Pull request merged**